src/base/bittorrent/session.cpp

```diff
@@ -3507,6 +3499,7 @@ void Session::handleTorrentResumeDataReady(TorrentHandle *const torrent, const l
     // isn't cheap too.
 
     QByteArray out;
+    out.reserve(10 * 1024 * 1024);  // most fastresume file sizes are under 10 MB
```

Comment:
It seems that usually these files are much smaller... Won't this cause a drop in performance?

Reply:
Here is one large fastresume: https://github.com/qbittorrent/qBittorrent/issues/11325#issuecomment-546604171

>Won't this cause a drop in performance?

Unless the machine comes with less memory, otherwise I think not (it merely reserve memory with allocator and move the pointer isn't ?).

Reply:
Allocating a longer contiguous memory area is harder, isn't it?
Plus, there's not much point in it if big files are rare and most of the time we have small files.

Reply:
>Allocating a longer contiguous memory area is harder, isn't it?

Yes.

>Plus, there's not much point in it if big files are rare and most of the time we have small files.

So what do you suggest?

Reply:
>So what value do you suggest?

Do nothing. Or reserve commonly used size.

On the other hand, if we constantly use some buffer, why should we allocate it every time? Would it be better to have some static buffer (i.e. class member)? In addition, in this case, it will automatically expand to the maximum file size and then will use it.

Reply:
>Do nothing. 

The file is always larger than a few bytes so some chosen value will always have a positive effect.

>Or reserve commonly used size.

You need a size that is **larger** than commonly used size to avoid relocations. Unless you suggest some I'll stay with the current.

>On the other hand, if we constantly use some buffer, why should we allocate it every time?

If qbt uses this buffer non-trivial amount of time (per lifetime) then it make sense, otherwise not.
Also reserve != allocate elements, AFAIK it is different to the memory system.

Reply:
>I think fastresume size depends on torrent properties/state so there is no unified size. Different users can have different max fastresume size.

Everything has some kind of distribution if given enough samples. Unless you imply we are facing an uniform distribution in this case (which I doubt), I'm not sure what you really implying.

Reply:
BTW, this change is out of PR topic, isn't it? Maybe move it into separate topic?

Reply:
>@thalieht
May I ask you to take a look at your fastresume files please? What is the biggest/average fastresume size you have? Mine max is ~100kB and on average ~40kB.

83K the largest and the majority of the others are less than 5K. Not sure what bloats .fastresume files but for the record, i don't rely on trackers and most of my torrents come with few/none.

Reply:
>BTW, this change is out of PR topic, isn't it? Maybe move it into separate topic?

OK, moved to #11422. Also lowered value to 100KB.

@thalieht 
Thanks!